### ISSUE TITLE
Support searching by ID

### ISSUE THREAD

**Issue by JackLCL** [milvus-io/milvus#3203]
As title.

**Comment by JinHai-CN** [comment]
detailed requirements needed.

**Comment by JackLCL** [comment]
In Milvus, each entity has a corresponding id. We need an interface to search for similar entities based on entity id.

**Comment by ufuk23** [comment]
We would like to search by given ids as well as vectors using rest API.

For example:

POST
http://localhost:19121/collections/art/vectors
{
"search": {
"topk": 3,
"vectors": [[0,0,0,0,0,0,1,0,0,0,0,0,0,0,0.01,0,0,0,0,0,0,0.2,0,0,0,0,0,0,0,0.3,0,0,0,0,0,0,0,0,0,0,0]],
"params": {"nprobe": 16},
"ids":["5", "23", "123", "1234", "334"]
}
}

For example, we need a scenario that works on Solr search engine and get the filtered results with IDs from Solr to search their feature vectors in Milvus by the same IDs. We have the filtered IDs coming from Solr and need to search their feature vectors. We don't want to search the whole collection of the milvus db, we need a subset of the collection by specified IDs.

**Comment by stephenleo** [comment]
I have a collection of 100Million `| product_id | vector |` in a collection.
I would like to find the top_k for any product within the above 100Million products.
Is there a simple way to do this?

For example:
My collection contains
```
| product_id_1 | vector_1 |
| product_id_2 | vector_2 |
| ...          | ...      |
| product_id_N | vector_N |
```
I want to find the top_k `product_ids` for `product_id_1`.

My current approach is use `get_entity_by_id(product_id_1)` to find out `vector_1` and then use `search(vector_1)` to get the top_k. Wondering if there is a simpler one step process?

Thanks

**Comment by jingkl** [comment]
For this question refer to the issue https://github.com/milvus-io/milvus/issues/4811

**Comment by yhmo** [comment]
There are two requirements here:
1. Input vectors as target nq, input some ids as filtering condition, and do similarity search. This feature has already been implemented in v2.0 RC4, this is the feature link: #5443

2. Input ids as target nq, to do similarity search. This feature is tracked by #5215
